lib/benchee/benchmark/scenario_context.ex

```diff
@@ -0,0 +1,21 @@
+defmodule Benchee.Benchmark.ScenarioContext do
+  defstruct [
```

Comment:
Seems like a smell to me :)

Also, on further inspection - why do we carry along the whole config when the only values we seem to be using are warmup, run_time and show_fast_warning? (If I didn't forget anyone)

We could just pick those values from the configuration and drop the configuration itself in the ScenarioContext imo. Anything speaking against this?

I somehow also don't feel like updating the ScenarioContext for the warmup is necessarily the best as I see it more immovable than this and a bit confusing (as it's the context for the whole scenario, if it changes based on warmup adn run time there might be another `ExecutionContext` but that seems like overkill). I feel like passing the benchmarking function the time it should run for would still be viable but I'd need to play with that to see how many arguments it adds where. Would like to play with that after this is merged, so definitely can go ahead with the ScenarioContext being adjusted. Would still like to explore dropping the dependency on configuration :)